src/Resources/contao/models/PageModel.php

```diff
@@ -981,11 +1005,63 @@ public function getFrontendUrl($strParams=null, $strForceLang=null)
 	 */
 	public function getAbsoluteUrl($strParams=null)
 	{
-		$strUrl = \Controller::generateFrontendUrl($this->loadDetails()->row(), $strParams, null, true);
+		$this->loadDetails();
 
-		if (strncmp($strUrl, 'http://', 7) !== 0 && strncmp($strUrl, 'https://', 8) !== 0)
+		$objUrlGenerator = \System::getContainer()->get('contao.routing.url_generator');
+
+		$strUrl = $objUrlGenerator->generate
+		(
+			($this->alias ?: $this->id),
+			array
+			(
+				'_locale' => $this->rootLanguage,
+				'_domain' => $this->domain,
+				'_ssl' => (bool) $this->rootUseSSL,
+			),
+			UrlGeneratorInterface::ABSOLUTE_URL
+		);
+
+		$strUrl = $this->applyLegacyLogic($strUrl, $strParams);
```

Comment:
There is no other way I guess if we want to use the router with the `UrlGeneratorInterface::ABSOLUTE_URL` constant. Otherwise we had to add the scheme, host and path manually.